python/pacemaker/_cts/tests/base.py

```diff
@@ -0,0 +1,915 @@
+""" Base classes for CTS tests """
+
+__all__ = ["CTSTest", "RemoteDriver", "SimulStartLite", "SimulStopLite", "StartTest", "StopTest"]
+__copyright__ = "Copyright 2000-2023 the Pacemaker project contributors"
+__license__ = "GNU General Public License version 2 or later (GPLv2+) WITHOUT ANY WARRANTY"
+
+#
+#        SPECIAL NOTE:
+#
+#        Tests may NOT implement any cluster-manager-specific code in them.
+#        EXTEND the ClusterManager object to provide the base capabilities
+#        the test needs if you need to do something that the current CM classes
+#        do not.  Otherwise you screw up the whole point of the object structure
+#        in CTS.
+#
+#                Thank you.
+#
```

Comment:
I'm thinking we can just drop this note. Hopefully once everything's better organized it will be more obvious where to do what. Alternatively we could add a section about CTS to Pacemaker Development and include it there.

Reply:
Fixed with an additional commit.